### PR TITLE
TC: Add test cases for Libc Fixedmath

### PR DESCRIPTION
**Pull request by lokeshbv**
1) Adds new Kconfig **TC_KERNEL_LIBC_FIXEDMATH** to enable tc for libc fixedmath.
2) Updates  tc **defconfig**s of **sidk_s5jt200** and **artik053** boards with new Kconfig.
3) Adds test cases for **libc/fixedmath** API's namely,
     **b16atan2, b16cos, b16divb16, b16mulb16, b16sin, b16sqr, ub16divub16, ub16mulub16 and ub16sqr**

### CODE REVIEW

File: build/configs/artik053/tc/defconfig [hunk 1]
```diff
@@ -1091,6 +1091,7 @@ CONFIG_TC_KERNEL_ENVIRON=y
 CONFIG_TC_KERNEL_ERRNO=y
 # CONFIG_TC_KERNEL_GROUP is not set
 CONFIG_TC_KERNEL_LIBC_LIBGEN=y
+CONFIG_TC_KERNEL_LIBC_FIXEDMATH=y
```
Comment (manoharahk) [inline build/configs/artik053/tc/defconfig]: It is better to split Kconfig & defconfig changes to separate  patches.
1. One Patch for New Kconfig,
2. Updating defconfig to enable "FixedMath"

Comment (lokeshbv) [inline build/configs/artik053/tc/defconfig]: okay done

File: apps/examples/testcase/le_tc/kernel/kernel_tc_main.c [hunk 2]
```diff
@@ -79,6 +79,10 @@ int kernel_tc_main(int argc, char *argv[])
 	libc_libgen_main();
 #endif
 
+#ifdef CONFIG_TC_KERNEL_LIBC_FIXEDMATH
+	libc_fixedmath_main();
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/kernel_tc_main.c]: check the alphabet sequence.

Comment (lokeshbv) [inline apps/examples/testcase/le_tc/kernel/kernel_tc_main.c]: changes made, please review the same

File: apps/examples/testcase/le_tc/kernel/tc_libc_fixedmath.c [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+static void tc_fixedmath_b16sin(void)
+{
+	int degree;
+	int radian;
+	int output;
+	b16_t b16radian;
+	b16_t b16output;
+
+	degree = -233;
+	radian = (degree * M_PI )/ 180;
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/tc_libc_fixedmath.c]: need space between ) and /.

File: apps/examples/testcase/le_tc/kernel/tc_libc_fixedmath.c [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+	output = (int)sinf(radian);
+
+	b16radian = itob16(radian);
+	b16output = b16sin(b16radian);
+	TC_ASSERT_EQ("b16sin", output, b16toi(b16output));
+
+	degree = 0;
+	radian = (degree * M_PI )/ 180;
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/tc_libc_fixedmath.c]: same as above

File: apps/examples/testcase/le_tc/kernel/tc_libc_fixedmath.c [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+static void tc_fixedmath_b16cos(void)
+{
+	int degree = 0;
+	int radian;
+	int output;
+	b16_t b16radian;
+	b16_t b16output;
+
+	radian = (degree * M_PI )/ 180;
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/tc_libc_fixedmath.c]: same

File: apps/examples/testcase/le_tc/kernel/tc_libc_fixedmath.c [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+	degree = -233;
+	radian = (degree * M_PI ) / 180;
+	output = (int)sinf(radian);
+
+	b16radian = itob16(radian);
+	b16output = b16sin(b16radian);
+	TC_ASSERT_EQ("b16sin", output, b16toi(b16output));
+
+	degree = 0;
+	radian = (degree * M_PI ) / 180;
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/tc_libc_fixedmath.c]: remove unncessary space

File: apps/examples/testcase/le_tc/kernel/tc_libc_fixedmath.c [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+static void tc_fixedmath_b16sin(void)
+{
+	int degree;
+	int radian;
+	int output;
+	b16_t b16radian;
+	b16_t b16output;
+
+	degree = -233;
+	radian = (degree * M_PI ) / 180;
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/tc_libc_fixedmath.c]: remove unncessary space

File: apps/examples/testcase/le_tc/kernel/tc_libc_fixedmath.c [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+
+static void tc_fixedmath_b16cos(void)
+{
+	int degree = 0;
+	int radian;
+	int output;
+	b16_t b16radian;
+	b16_t b16output;
+
+	radian = (degree * M_PI ) / 180;
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/tc_libc_fixedmath.c]: remove unncessary space

**Comment by lokeshbv** [comment]
@jeongchanKim are you talking about, adding new config or is there any config that can be bind here?

**Comment by jeongchanKim** [comment]
I mean CONFIG_LIBM

**Comment by lokeshbv** [comment]
But, the libc/fixedmath is not bound by any config. is it necessary here?

**Comment by jeongchanKim** [comment]
yes, they are not binded because user can choose which math API use, libc or architecture based or something.
If you enable CONFIG_LIBM, you can use libc API. And now you are try to verify the libc API.
You can refer to os/Makefile.unix

**Comment by sunghan-chang** [comment]
Testcase is for verification of our codes, libc. If there is no dependancy, user can execute tc with arch math or compiler math. And if someone meets errors on these tc, it is difficult to find and we don't want to fix it. That's why we recommand binding LIBM conditional.

**Comment by lokeshbv** [comment]
Ok, got it. I will bind LIBM conditional for Tcs.

**Comment by lokeshbv** [comment]
@sunghan-chang, @jeongchanKim, Is the following binding of **CONFIG_LIBM** in **_apps/examples/testcase/le_tc/kernel/Make.defs_** is fine?

```
ifeq ($(CONFIG_LIBM), y)
ifeq ($(CONFIG_TC_KERNEL_LIBC_FIXEDMATH),y)
  CSRCS += tc_libc_fixedmath.c
endif
endif
```

File: apps/examples/testcase/le_tc/kernel/Kconfig [hunk 9]
```diff
@@ -67,6 +68,10 @@ config TC_KERNEL_GROUP
 	select SCHED_HAVE_PARENT
 	select SCHED_CHILD_STATUS
 
+config TC_KERNEL_LIBC_FIXEDMATH
+	bool "Libc FixedMath"
+	default n
+
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/Kconfig]: add a dependancy here.
```
config TC_KERNEL_LIBC_FIXEDMATH
	bool "Libc FixedMath"
	default n
	depends on LIBM
```

**Comment by lokeshbv** [comment]
@sunghan-chang , Added LIBM dependency to Libc FixedMath config.

**Pull request merged**